src/dialogs/components/dialog/dialog.ts

```diff
@@ -0,0 +1,143 @@
+       this.service = this.DialogData;
+       for (var dialogTabs of this.dialog.dialog_tabs){
+            for (var dialogGroup of dialogTabs.dialog_groups){
+                for (var dialogField of dialogGroup.dialog_fields) {
+                    dialogField.fieldValidation='';
+                    vm.dialogFields[dialogField.name] = dialogField; // at this point all dialog fields are stored in a object keyed by field name
+                    vm.dialogValues[dialogField.name] = '';
+
+                    if (dialogField.auto_refresh === true || dialogField.trigger_auto_refresh === true) {
+                        vm.refreshableFields.push(dialogField.name);
+                    }
+                }
+            }
+       }
+    }
+  /**
+   * This reports all values from the dialog up to the parent component
+   * The onUpdate method signature from the parent component should be updateFunctionName(data)
+   *
+   * saveDialogData
+   * @memberof DialogController
+   * @function saveDialogData
+   */
+    public saveDialogData () {
+      const outputData = {
+        validations: {},
+        data: this.dialogValues
+      };
+      this.onUpdate({data: outputData});
+    }
+    /**
+     * This method handles refreshing of a dialog field as well as determining wish other fields might need to be updated
+     * @function updateDialogField
+     * @param dialogFieldName {string} This is the field name for the particular dialog field
+     * @param value {any} This is the updated value based on the selection the user made on a particular dialog field 
+     */
+    public updateDialogField(dialogFieldName, value) {
+        const refreshable = _.indexOf(this.refreshableFields,dialogFieldName);
+        this.dialogFields[dialogFieldName].default_value = value;
+        this.dialogValues[dialogFieldName] = value;
+
+        if (refreshable > -1) {
+            this.refreshField({ field: this.dialogFields[dialogFieldName] }).then((data) => {
+                this.dialogFields[dialogFieldName] = this.updateDialogFieldData(dialogFieldName, data);
+                this.updateRefreshableFields(dialogFieldName);
+            });
+        }
+        this.saveDialogData();
+    }
+/**
+ * This method is meant to handle auto updating of all dialog fields that are eligable to be refreshed after a field has just been refreshed
+ *  @function updateRefreshableFields
+ *  @param triggerFieldName {string} This is the dialog fields name that was triggered.  This is passed to ensure we don't attempt to refresh something that was just refreshed
+ */
+    public updateRefreshableFields(triggerFieldName){
+       const fieldsToRefresh = _.without(this.refreshableFields, triggerFieldName);
+       fieldsToRefresh.forEach((field) => {
+            this.refreshField({ field: this.dialogFields[field] }).then((data) => {
+                this.dialogFields[field] = this.updateDialogFieldData(field, data);
+            });
+       });
+    }
```

Comment:
If I understand this correctly, this would change the way cascading currently works, right? This is going to loop through all the fields to refresh and immediately fire off the `.refreshField` function, . If, in the list of `fieldsToRefresh`, the 2nd one depends on the 1st one completing, this will not behave like it does currently.

I think instead of iterating, we need to do something recursive like:
```typescript
public updateRefreshableFields(listOfFieldsToExclude) {
  const fieldsToRefresh = _.without(this.refreshableFields, listOfFieldsToExclude);
  return if fieldsToRefresh[0] === undefined;

  const firstFieldToRefresh = fieldsToRefresh[0];
  const listOfFieldsToExclude = listOfFieldsToExclude.concat(firstFieldToRefresh);

  firstFieldToRefresh.refreshField({ field:this.dialogFields[firstFieldToRefresh] }).then(data) => {
    this.dialogFields[firstFieldToRefresh] = this.updateDialogFieldData(firstFieldToRefresh, data);
    this.updateRefreshableFields(listOfFieldsToExclude);
  });
}
```